### PR TITLE
Hotfix/translation update tool

### PR DESCRIPTION
**Pull request by phoerious**
## Description

This patch improves the `update.sh` script under `share/translations` to allow updating translation source files, pushing or pulling in translations without the need to perform a full update/push/pull cycle.

The update script now has three subcommands:

- `push` update source files and push to Transifex
- `pull` pull in translations from Transifex
- `update` only update source files, don't push or pull anything

If no subcommand was specified, all three actions are performed.

## Motivation and context

Pushing translations to Transifex only when bundling the release is quite bad practice. That way a new release will always have untranslated strings, because translators hadn't had the chance to translate anything. Ideally, we should periodically push any new translation changes to Transifex, either as separate pull requests or as addendum to other pull requests. This patch makes that possible. Of course, one can also just update and push and then `git reset` their working tree (which is basically what we have done so far).

Pulling translations can still be done during release building.

## How has this been tested?

I was able to run all three commands separately. Running the script without a subcommand still works, too.
Updated source files are appended to this PR.

## Types of changes

- ✅ New feature (non-breaking change which adds functionality)

## Checklist:

- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**

### CODE REVIEW

**Comment by droidmonkey** [comment]
Long time coming, great work

File: share/translations/update.sh [hunk 1]
```diff
@@ -1,17 +1,58 @@
...
-echo Pushing English translation file to Transifex
-tx push -s
+cd "${BASEDIR}/../.."
 
-echo
-echo Pulling translations from Transifex
-tx pull -af --minimum-perc=40
+if $UPDATE; then
+    echo "Updating source files..."
+    lupdate-qt5 -no-ui-lines -disable-heuristic similartext -locations none -no-obsolete src -ts share/translations/keepassx_en.ts
```
Comment (droidmonkey) [inline share/translations/update.sh]: Sometimes lupdate-qt5 is actually in the system as simply lupdate. Perhaps the script can detect between the two and smartly choose the right one.

Comment (phoerious) [inline share/translations/update.sh]: Sure. Will update later.

Comment (phoerious) [inline share/translations/update.sh]: Done.

**Pull request merged**